### ISSUE TITLE
Themable Default Colors

### ISSUE THREAD

**Issue by Unthrottled** [krasa/GrepConsole#205]
**Is your feature request related to a problem? Please describe.**

The plugin default highlighting feature doesn't always look the greatest for custom UI themes. 

**Describe the solution you'd like**
Allow the default background/foreground colors to be customizable by allowing UI themes to provide a default value in place.

**Describe alternatives you've considered**
I could manually update the colors, but I don't what to have to do that every time I change a theme. Would be nice if it could change with the UI theme.

**Additional context**

What I am looking for is like how a theme can specify `FileColor.Blue` as a LaF property and it overrides the default for the blue setting. 

![image](https://user-images.githubusercontent.com/15972415/147238601-6a7e5f9b-353e-4b59-9e25-a59c4ec8196a.png)

**Comment by Unthrottled** [comment]
This is also something I could add, just wanted to make sure you would be cool with it.

**Comment by krasa** [comment]
Sure, go ahead.

**Comment by Unthrottled** [comment]
So after looking into adding this issue, I found out the colors themselves are serialized into storage for later use. Which is inconvenient, if I want the value to change with the look and feel like JBColor.

I think I might be better off writing an integration with this plugin which just automatically sets the colors & re-highlightes, than trying to add this feature in. I would like to not have to make large changes.

Unless I am missing something simple.

**Comment by krasa** [comment]
JBColor was not used because its colors are not mapped well for this purpose. 

Imho GrepColor could have a map of custom colors for each theme (Map<String, Color>) + ColorKey for default values, deprecating GrepColor.color. Then your theme could override that ColorKey. 
It should be easy, with minor changes to CheckBoxWithColorChooser and DefaultState#createDefaultItems. 
Then I could remove resolving of groups based on light/dark theme, which would not be needed anymore.
It sounds to me like something I should have done already.

Alternatively:
Adding resolving of groups based on a theme name could help a little, that would be super easy, but maybe the least useable.

**Comment by Unthrottled** [comment]
I see, so instead of persisting the actual color for a profile, the chosen ColorKey will somehow be persisted. That is assuming that the names you had in mind are like the name of colors like the dropdown in the original post (`Rose`, `Violet`, etc).

One of the things that I did run into was that I had to trigger a ReHighlight for the change in color to be picked up when the LaF changed. (I've got a working integration now.)

If I am understanding things correctly:

The com.intellij.openapi.editor.colors.ColorKey concept correctly, define a set of color keys. Allow the user to save which color key they want to use for each highlight. Provide default values for each color key (light and dark). I just don't how much effort providing a custom option would be.
